src/main/java/io/stargate/sgv2/jsonapi/service/resolver/CommandResolver.java

```diff
@@ -213,4 +216,24 @@ private void getIndexUsageTags(DBLogicalExpression dbLogicalExpression, IndexUsa
       getIndexUsageTags(subDBLogicalExpression, indexUsage);
     }
   }
+
+  /**
+   * Validate the name of a schema object (e.g., keyspace, table, collection, or index).
+   *
+   * @param name The name to validate.
+   * @param namingRule The naming rule to apply.
+   * @throws SchemaException if the name is invalid.
+   */
+  default void validateSchemaName(String name, SchemaObjectNamingRule namingRule) {
+    if (!namingRule.apply(name)) {
+      throw SchemaException.Code.UNSUPPORTED_SCHEMA_NAME.get(
+          Map.of(
+              "schemaType",
+              namingRule.name(),
+              "maxNameLength",
+              String.valueOf(SchemaObjectNamingRule.MAX_NAME_LENGTH),
+              "unsupportedSchemaName",
+              name == null ? "null" : name));
```

Comment:
Need to use something else for `null` case since "null" would be legal table/keyspace etc name? Maybe `<null>`?

Reply:
Also, I noticed (see a later comment) that error messages look quite strange if name is empty String (or all white-space).

I think names in general should be quoted so they are easier to tell from messages.
So maybe define `quote(String name)` method that does something like:

```java
public static String quote(String name) {
     if (name == null) {
        return "<null>";
    }
   return "'%s'".formatted(name);
}
```

called from here.